vividus/src/main/java/org/vividus/configuration/PropertiesDecryptor.java

```diff
@@ -38,11 +38,15 @@ public static Properties decryptProperties(StringEncryptor stringEncryptor, Prop
             Object value = entry.getValue();
             if (value instanceof String)
             {
-                Matcher matcher = ENCRYPTED_PROPERTY_PATTERN.matcher((String) value);
-                if (matcher.matches())
+                String decryptedValue = (String) value;
+                Matcher matcher = ENCRYPTED_PROPERTY_PATTERN.matcher(decryptedValue);
+                while (matcher.matches())
```

Comment:
why not to keep ENCRYPTED_PROPERTY_PATTERN as it was before and use find() instead of matches()?